### PR TITLE
redis: implement XADD stream support - v2

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #11294 

Link to ticket: https://redmine.openinfosecfoundation.org/issues/7082

Describe changes to previous PR:
- Add support for automatic trimming via `XADD`. See https://github.com/OISF/suricata/pull/11294#issuecomment-2387895092 for more details.

### CODE REVIEW

File: doc/userguide/output/eve/eve-json-output.rst [hunk 1]
```diff
@@ -41,10 +41,18 @@ Output types::
...
       #  port: 6379
       #  async: true ## if redis replies are read asynchronously
-      #  mode: list ## possible values: list|lpush (default), rpush, channel|publish
+      #  mode: list ## possible values: list|lpush (default), rpush, channel|publish, xadd|stream
       #             ## lpush and rpush are using a Redis list. "list" is an alias for lpush
       #             ## publish is using a Redis channel. "channel" is an alias for publish
-      #  key: suricata ## key or channel to use (default to suricata)
+      #             ## xadd is using a Redis stream. "stream" is an alias for xadd
+      #  key: suricata ## string denoting the key/channel/stream to use (default to suricata)
+      #  stream-maxlen: 10000000      ## Automatically trims the stream length to at most
```
Comment (jasonish) [inline doc/userguide/output/eve/eve-json-output.rst:49]: Seems large default a default? Maybe 100000?

File: doc/userguide/output/eve/eve-json-output.rst [hunk 2]
```diff
@@ -41,10 +41,18 @@ Output types::
...
+                                      ## this number of events. Set to 0 to disable trimming.
+                                      ## Only used when mode is set to xadd/stream.
+      #  stream-trim-exact: False     ## Trim exactly to the maximum stream length above.
```
Comment (jasonish) [inline doc/userguide/output/eve/eve-json-output.rst:52]: Should be `false`.  `False` is still falsy, but `True` is not `true`.

**Review by jasonish — CHANGES_REQUESTED** [review]
Minor nits, but tests OK.

File: doc/userguide/output/eve/eve-json-output.rst (hunk 2 above)
Comment (satta) [reply doc/userguide/output/eve/eve-json-output.rst:52]: Right, will do.

File: doc/userguide/output/eve/eve-json-output.rst (hunk 1 above)
Comment (victorjulien) [reply doc/userguide/output/eve/eve-json-output.rst:49]: unbounded sounds kind of bad, what effects are there if we keep growing?

Comment (satta) [reply doc/userguide/output/eve/eve-json-output.rst:49]: In that case the Redis server will have to handle the size of the stream, e.g. by evicting items once memory fills up. IMHO it is up to the admin of the setup (who knows the access pattern) to maintain a safe stream size -- similar to how a Redis list (`redis.mode: list`) can also grow indefinitely if nothing is popped from it.

However, to be on the safe side I am going to change this setting to be 100K as default and leave it to the user to make it larger or set it to unbounded.

**Comment by satta** [comment]
Next PR: #12003

File: doc/userguide/output/eve/eve-json-output.rst (hunk 1 above)
Comment (victorjulien) [reply doc/userguide/output/eve/eve-json-output.rst:49]: What will suri do if this limit is reached?

Comment (jasonish) [reply doc/userguide/output/eve/eve-json-output.rst:49]: > What will suri do if this limit is reached?

Nothing. When using max len, Redis will evict the oldest data.

Comment (satta) [reply doc/userguide/output/eve/eve-json-output.rst:49]: Correct.

Comment (victorjulien) [reply doc/userguide/output/eve/eve-json-output.rst:49]: Ok, thanks.

**Pull request closed without merging**